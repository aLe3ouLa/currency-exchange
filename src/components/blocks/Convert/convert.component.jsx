import React from "react";

import FormInput from "../../fragment/FormInput/form-input.component";
import Currencies from "../../fragment/Currencies/currencies.component";
import Balance from "../../fragment/Balance/balance.component";

import { Container, FormWrapper, CurrencyCtn } from "./convert.styles";

const Convert = ({
  balance,
  fromValue,
  toValue,
  toCurrency,
  fromCurrency,
  isFromValue,
  rates: { rates },
  handleChange,
  handleCurrencyChange,
}) => {
  const value = isFromValue ? fromValue : toValue;
  const currency = isFromValue ? fromCurrency : toCurrency;
  return (
    <Container isFromValue={isFromValue}>
      <CurrencyCtn>
        <Currencies
          name={currency}
          value={currency}
          onChange={handleCurrencyChange}
          isDisabled={false}
        />
        <FormWrapper>
          <FormInput
            name="value"
            value={value}
            onChange={handleChange}
            disabled={!rates.date}
            isFromValue={isFromValue}
            isZero={value === 0}
          />
        </FormWrapper>
      </CurrencyCtn>
      <Balance
        balance={balance}
        currency={currency}
        fromValue={fromValue}
        isFromValue={isFromValue}
      />
    </Container>
  );
};

export default Convert;
