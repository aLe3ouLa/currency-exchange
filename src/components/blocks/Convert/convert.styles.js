import styled, { css } from "styled-components";

export const Container = styled.div`
  height: 50%;
  box-sizing: border-box;
  padding: 20px;
  ${({ isFromValue }) =>
    isFromValue
      ? css`
          background-color: ${({ theme }) => theme.color.white};
        `
      : css`
          background-color: ${({ theme }) => theme.color.background};
          height: 100% !important;
        `}
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const FormWrapper = styled.div`
  display: flex;
  justify-content: flex-end;
  flex-basis: 100%;
`;
