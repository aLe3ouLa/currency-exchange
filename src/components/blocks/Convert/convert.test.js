import React from "react";
import ReactDOM from "react-dom";
import Convert from "./convert.component";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";
describe("Convert tests", () => {
  afterEach(cleanup);

  let props = {
    rates: {
      base: "GBP",
      rates: {
        EUR: 12.01,
        GBP: 0.0,
        USD: 10.01,
      },
      date: new Date(),
    },
    active: {
      fromCurrency: "EUR",
      fromValue: 1,
      toCurrency: "GBP",
      toValue: 1,
    },
    balance: {
      EUR: 12.01,
      GBP: 1.0,
      USD: 10.01,
      JPY: 11.0,
      lastExchange: {
        fromCurrency: "GBP",
        fromValue: 1,
        toCurrency: "EUR",
        toValue: 1,
        date: new Date(),
      },
    },
    isLoading: false,
  };

  it("renders the Convert component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Convert
            balance={props.balance}
            toValue={props.active.toValue}
            toCurrency={props.active.toCurrency}
            fromCurrency={props.active.fromCurrency}
            fromValue={props.active.fromValue}
            rates={props.rates}
            isFromValue={false}
            handleChange={jest.fn()}
            handleCurrencyChange={jest.fn()}
          />
        </React.Fragment>
      </ThemeProvider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Convert
            balance={props.balance}
            toValue={props.active.toValue}
            toCurrency={props.active.toCurrency}
            fromCurrency={props.active.fromCurrency}
            fromValue={props.active.fromValue}
            rates={props.rates}
            isFromValue={false}
            handleChange={jest.fn()}
            handleCurrencyChange={jest.fn()}
          />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
