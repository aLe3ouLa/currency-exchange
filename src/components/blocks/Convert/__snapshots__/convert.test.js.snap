// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rates tests has a valid snapshot  1`] = `
<DocumentFragment>
  <div
    class="sc-AxhUy kmTHDE"
  >
    <div
      style="display: flex; align-items: center;"
    >
      <select
        class="sc-AxiKw dhSpfk"
        data-testid="select"
        id="GBP"
        name="GBP"
      >
        <option
          label="GBP"
          value="GBP"
        />
        <option
          label="EUR"
          value="EUR"
        />
        <option
          label="USD"
          value="USD"
        />
        <option
          label="JPY"
          value="JPY"
        />
      </select>
      <div
        class="sc-AxgMl IvxZu"
      >
        <span
          class="sc-AxirZ kBcKOJ"
        >
          +
        </span>
        <input
          aria-label="form-input"
          autocomplete="off"
          class="sc-AxjAm ibqwxn"
          disabled=""
          name="value"
          placeholder="0"
          type="number"
          value="1"
        />
      </div>
    </div>
    <p
      class="sc-AxhCb ipKkGq"
    >
      Balance: Â£ 1
    </p>
  </div>
</DocumentFragment>
`;
